prymer/offtarget/bwa.py

```diff
@@ -357,17 +360,10 @@ def _to_result(self, query: Query, rec: pysam.AlignedSegment) -> BwaResult:
                 "Query and Results are out of order" f"Query=${query.id}, Result=${rec.query_name}"
             )
 
-        num_hits: Optional[int] = int(rec.get_tag("HN")) if rec.has_tag("HN") else None
-        if rec.is_unmapped:
-            if num_hits is not None and num_hits > 0:
-                raise ValueError(f"Read was unmapped but num_hits > 0: {rec}")
-            return BwaResult(query=query, hit_count=0, hits=[])
-        elif num_hits > self.max_hits:
-            return BwaResult(query=query, hit_count=num_hits, hits=[])
-        else:
-            hits = self.to_hits(rec=rec)
-            hit_count = num_hits if len(hits) == 0 else len(hits)
-            return BwaResult(query=query, hit_count=hit_count, hits=hits)
+        num_hits: int = int(rec.get_tag("HN")) if rec.has_tag("HN") else 0
+        hits: list[BwaHit] = self.to_hits(rec=rec) if 0 < num_hits <= self.max_hits else []
```

Comment:
Someone check my logic here - this may need to be `< self.max_hits`, because `self.max_hits` may refer to the maximum number of **secondary** hits in the XA tag (i.e. the `bwa samse -n` parameter), not the maximum number of **total** hits (which would be the number of secondary + 1)

Reply:
Callout from Alison - this should instead be:
- empty list when `num_hits=0`
- list containing all hits when `num_hits` is between 0 and max
- list containing only the primary alignment when `num_hits` is greater than (or equal to?) max

tbh I think this logic probably belongs in `to_hits()`, not here

Reply:
@tfenne Could you please weigh in on whether the list of hits should include the primary alignment or be empty when `num_hits` is greater than max?